src/Workspaces/Core/Portable/Classification/ClassifierHelper.cs

```diff
@@ -105,29 +109,33 @@ private static void RemoveAdditiveSpans(ArrayBuilder<ClassifiedSpan> spans)
             AdjustSpans(syntaxSpans, widenedSpan);
             AdjustSpans(semanticSpans, widenedSpan);
 
+            using var _1 = ArrayBuilder<ClassifiedSpan>.GetInstance(out var mergedSpans);
```

Comment:
core log changed in two ways.  First, we always merge syntactic and semantic first.  *Then* we fill in the spans in the gaps if requested.

Second, when doing the merge, the algorithm is redone to make it simpler and safer.